.github/workflows/release-python.yml

```diff
@@ -62,20 +70,224 @@ jobs:
+            openmp-11.0.0.src/build-arm64/install/lib/libomp.a \
+            -o $GITHUB_WORKSPACE/openmp-install/lib/libomp.a
+
+    - name: install-openmp for macos
+      if: matrix.os == 'macos-latest'
+      run: |
+        sudo cp $GITHUB_WORKSPACE/openmp-install/include/* $DEVELOPER_DIR/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include
+        sudo cp $GITHUB_WORKSPACE/openmp-install/lib/libomp.a $DEVELOPER_DIR/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib
+
+    - name: vulkansdk for macos
+      if: matrix.os == 'macos-latest'
+      run: |
+        wget https://sdk.lunarg.com/sdk/download/1.3.236.0/mac/vulkansdk-macos-1.3.236.0.dmg?Human=true -O vulkansdk-macos-1.3.236.0.dmg
+        hdiutil attach vulkansdk-macos-1.3.236.0.dmg
+        sudo /Volumes/vulkansdk-macos-1.3.236.0/InstallVulkan.app/Contents/MacOS/InstallVulkan --root $GITHUB_WORKSPACE/vulkansdk-macos-1.3.236.0 --accept-licenses --default-answer --confirm-command install
+        hdiutil detach /Volumes/vulkansdk-macos-1.3.236.0
+
+    - name: Build wheels for macos x86_64
+      if: matrix.os == 'macos-latest' && matrix.arch == 'x86_64'
       uses: pypa/cibuildwheel@v2.16.2
       env:
         CIBW_ARCHS_MACOS: ${{ matrix.arch }}
-        CIBW_ARCHS_LINUX: ${{ matrix.arch }}
-        CIBW_ARCHS_WINDOWS: ${{ matrix.arch }}
         CIBW_BUILD: ${{ matrix.build }}
         CIBW_BUILD_VERBOSITY: 1
-        CIBW_ENVIRONMENT: CMAKE_BUILD_PARALLEL_LEVEL=2
+        CIBW_ENVIRONMENT: CMAKE_BUILD_PARALLEL_LEVEL=2 NCNN_VULKAN=ON VULKAN_SDK=$GITHUB_WORKSPACE/vulkansdk-macos-1.3.236.0/macOS
+          CMAKE_TOOLCHAIN_FILE=$GITHUB_WORKSPACE/toolchains/ios.toolchain.cmake PLATFORM=MAC ARCHS="x86_64"
+          DEPLOYMENT_TARGET="10.9" ENABLE_BITCODE=OFF ENABLE_ARC=OFF ENABLE_VISIBILITY=OFF
+          OpenMP_C_FLAGS="-Xclang -fopenmp" OpenMP_CXX_FLAGS="-Xclang -fopenmp"
+          OpenMP_C_LIB_NAMES="libomp" OpenMP_CXX_LIB_NAMES="libomp"
+          OpenMP_libomp_LIBRARY="$DEVELOPER_DIR/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib/libomp.a"
+          Vulkan_INCLUDE_DIR=$GITHUB_WORKSPACE/vulkansdk-macos-1.3.236.0/MoltenVK/include
+          Vulkan_LIBRARY=$GITHUB_WORKSPACE/vulkansdk-macos-1.3.236.0/MoltenVK/dylib/macOS/libMoltenVK.dylib
+      with:
+        output-dir: wheelhouse
+
+    - name: Build wheels for macos arm64 and macos universal2
+      if: (matrix.os == 'macos-latest' && matrix.arch == 'arm64') ||
+        (matrix.os == 'macos-latest' && matrix.arch == 'universal2')
+      uses: pypa/cibuildwheel@v2.16.2
+      env:
+        CIBW_ARCHS_MACOS: ${{ matrix.arch }}
+        CIBW_BUILD: ${{ matrix.build }}
+        CIBW_BUILD_VERBOSITY: 1
+        CIBW_ENVIRONMENT: CMAKE_BUILD_PARALLEL_LEVEL=2 NCNN_VULKAN=ON VULKAN_SDK=$GITHUB_WORKSPACE/vulkansdk-macos-1.3.236.0/macOS
+          CMAKE_TOOLCHAIN_FILE=$GITHUB_WORKSPACE/toolchains/ios.toolchain.cmake PLATFORM=MAC_ARM64 ARCHS="arm64"
```

Comment:
but universal build only contains arm64 arch  :(

Reply:
Some revision and tests are needed. I will have a try this weekend.

Reply:
"ios.toolchain.cmake" offers the MAC_UNIVERSAL options, it may be solved soon. 😄

Reply:
[7c1674b](https://github.com/Tencent/ncnn/pull/5020/commits/7c1674b5bf87826e129d557693807ef04d009388) is built with -target arm64-x86_64-apple-macosx11.0, it should be a 'real' universal2 wheel. The release test for it is:
https://github.com/Hideousmon/ncnn/actions/runs/6602816382
But I can not test it on a machine with gpu, when I try to install the universal2 wheel on my x86_64 macos (macOS 14.0, with pip==23.3.1), it is told that `ERROR: ncnn-1.0.20231022-cp38-cp38-macosx_11_0_universal2.whl is not a supported wheel on this platform`.

Reply:
Test it with a universal2 python on my x86_64 macos with gpu:
```
zz@zzs-MacBook-Pro bin % pip3 install ~/Downloads/ncnn-1.0.20231022-cp38-cp38-macosx_11_0_universal2.whl
zz@zzs-MacBook-Pro ncnnx86_64test % python3.8 yolov4-tiny-testbench.py
[0 Intel(R) Iris(TM) Plus Graphics]  queueC=0[1]  queueG=0[1]  queueT=0[1]
[0 Intel(R) Iris(TM) Plus Graphics]  bugsbn1=0  bugbilz=0  bugcopc=0  bugihfa=0
[0 Intel(R) Iris(TM) Plus Graphics]  fp16-p/s/a=1/1/1  int8-p/s/a=1/1/1
[0 Intel(R) Iris(TM) Plus Graphics]  subgroup=32  basic/vote/ballot/shuffle=1/1/1/1
[0 Intel(R) Iris(TM) Plus Graphics]  fp16-matrix-16_8_8/16_8_16/16_16_16=0/0/0
NCNN Init time: 521.049342 ms
NCNN Inference time: 112.29334700000004 ms
NCNN Detection time: 113.453374 ms
```
It is functional ! 😆